common/shared/src/main/java/app/coronawarn/server/common/shared/util/SerializationUtils.java

```diff
@@ -88,8 +94,31 @@ public static <T> Optional<T> deserializeJsonToSimpleType(InputStream jsonStream
   }
 
   /**
-   * Reads and convers a JSON object from a classpath file or if it does not find it
-   * returns a default.
+   * Extracts JSONObject from COSE byte sequence.
+   *
+   * @param byteSequence payload to be parsed
+   * @return parsed payload as JSON object
+   * @throws IOException    if parser cannot be created
+   * @throws ParseException if byte sequence cannot be parsed
+   */
+  public static Map<byte[], List<byte[]>> jsonExtractCosePayload(byte[] byteSequence)
+      throws IOException, ParseException {
+
+    CBORObject cborObject = CBORObject.DecodeFromBytes(byteSequence);
+    CBORObject payload = CBORObject.DecodeFromBytes(cborObject.get(2).GetByteString());
+
+    Map<byte[], List<byte[]>> payloadEntries = new HashMap<>();
+    payload.getKeys().forEach(key -> {
+      List<byte[]> values = payload.get(key).getValues().stream()
+          .map(value -> value.GetByteString()).collect(Collectors.toList());
```

Comment:
`byte[]` as key in the map is correct?

Reply:
yes, we parse kid byte array as the key in the map, and the List<byte[]> would be the hashes for the corresponding kid